### PR TITLE
chore: Add basic tests

### PR DESCRIPTION
**Pull request by caalador**
Make multiple test modules into integration-tests
Add productionMode test.
Add developmentMode test.
Add test for reusable theme.

part of #4

### CODE REVIEW

File: integration-tests/common-test-code/pom.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>3.0.0-M5</version>
+                <configuration>
+                    <skipTests>true</skipTests>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (denis-anisimov) [inline integration-tests/common-test-code/pom.xml:71]: new line

File: integration-tests/development/pom.xml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                                    <version>1.2.0</version>
+                                </dependency>
+                            </dependencies>
+                        </plugin>
+                    </plugins>
+                </pluginManagement>
+            </build>
+        </profile>
+    </profiles>
+</project>
```
Comment (denis-anisimov) [inline integration-tests/development/pom.xml:236]: new line

File: integration-tests/production/pom.xml [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                                    <version>1.2.0</version>
+                                </dependency>
+                            </dependencies>
+                        </plugin>
+                    </plugins>
+                </pluginManagement>
+            </build>
+        </profile>
+    </profiles>
+</project>
```
Comment (denis-anisimov) [inline integration-tests/production/pom.xml:234]: new line

File: integration-tests/reusable-theme/src/main/resources/META-INF/resources/themes/reusable-theme/styles.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+   This is the style entrypoint for the theme and together with css in ./components/ included
+   automatically into the theme.
+   Visit https://vaadin.com/docs-beta/latest/theming/application-theme/ for more information.
+*/
+
+/* Example: CSS class name to center align the content . */
+.centered-content {
+    margin: 0 auto;
+    max-width: 250px;
+}
```
Comment (denis-anisimov) [inline integration-tests/reusable-theme/src/main/resources/META-INF/resources/themes/reusable-theme/styles.css:13]: new line

**Review by denis-anisimov — CHANGES_REQUESTED** [review]

**Comment by caalador** [comment]
New lines added.

**Review by denis-anisimov — APPROVED** [review]

**Pull request merged**